lint: parenthesis suggests to simplify size expressions inside array types

```
cmd/internal/obj/x86/asm6.go:2043:14: parenthesis: could simplify (ALAST + 1) to ALAST + 1
```

`[(A + B)]int` is an array of ints with size `A+B`. 
Expressions can be complex and it's not quite right to trigger on them.

I don't remember any other cases except for array types, so maybe all what is required is to avoid array type length checks.

```go
	ArrayType struct {
		Lbrack token.Pos // position of "["
		Len    Expr      // Ellipsis node for [...]T array types, nil for slice types
		Elt    Expr      // element type
	}
```

`parenthesis` should not visit `ArrayType.Len`.

--- a/cmd/kfulint/testdata/parenthesis/checker_tests.go
+++ b/cmd/kfulint/testdata/parenthesis/checker_tests.go
@@ -31,7 +31,7 @@ var (
 ///: could simplify (int) to int
 const _ (int) = 5
 
-//TODO: could simplify (int) to int
+///: could simplify (int) to int
 type _ (int)
 
 const ALAST = 1

--- a/cmd/kfulint/testdata/parenthesis/checker_tests.go
+++ b/cmd/kfulint/testdata/parenthesis/checker_tests.go
@@ -36,9 +36,10 @@ type _ (int)
 
 const ALAST = 1
 
-//TODO: shouldn't create warning
-///: could simplify (ALAST + 1) to ALAST + 1
 var opindex [(ALAST + 1) & 3]*int
+
+///: could simplify (ALAST + 100 - 20*5) to ALAST + 100 - 20*5
+var _ [(ALAST + 100 - 20*5)]*int
 
 var _ func(int, string)
 

--- a/lint/parenthesis_checker.go
+++ b/lint/parenthesis_checker.go
@@ -57,15 +57,29 @@ func (c *ParenthesisChecker) validateType(n ast.Node) {
 	// TODO improve linter output to write full type, not just place
 	// where it could be simplified
 	ast.Inspect(n, func(n ast.Node) bool {
+		if n, ok := n.(*ast.ArrayType); ok {
+			c.validateType(n.Elt)
+
+			if expr, ok := n.Len.(*ast.ParenExpr); ok {
+				c.warn(expr)
+			}
+			return false
+		}
+
 		expr, ok := n.(*ast.ParenExpr)
 		if !ok {
 			return true
 		}
-		c.ctx.addWarning(Warning{
-			Kind: "parenthesis",
-			Node: expr,
-			Text: fmt.Sprintf("could simplify %s to %s", nodeString(c.ctx.FileSet, expr), nodeString(c.ctx.FileSet, expr.X)),
-		})
+		c.warn(expr)
 		return false
 	})
+}
+
+func (c *ParenthesisChecker) warn(expr *ast.ParenExpr) {
+	c.ctx.addWarning(Warning{
+		Kind: "parenthesis",
+		Node: expr,
+		Text: fmt.Sprintf("could simplify %s to %s", nodeString(c.ctx.FileSet, expr), nodeString(c.ctx.FileSet, expr.X)),
+	})
+
 }

--- a/lint/parenthesis_checker.go
+++ b/lint/parenthesis_checker.go
@@ -44,9 +44,13 @@ func (c *ParenthesisChecker) Check(f *ast.File) {
 						continue
 					}
 					c.validateType(spec.Type)
-					continue
 				}
-				// TODO add check for token.TYPE
+				if spec, ok := spec.(*ast.TypeSpec); ok {
+					if spec.Type == nil {
+						continue
+					}
+					c.validateType(spec.Type)
+				}
 			}
 		}
 	}

